README.md

```diff
@@ -698,3 +698,47 @@ module.settings.changeEdition = function (options, callback) {
   return callback(error, response)
 }
 ```
+
+#### getMappingMetadata
+getMappingMetadata function is used to add mapping validation for connectors to ensure certain business critical mappings are not removed/edited by a business user. This function is called on page refresh/load to get the mapping metadata for connector stack to make mappings non-editable or required (non-deletable) or both for a front end user.
+
+```js
+/*
+ * options object contains the following properties:
+ *     
+ *  bearerToken - a one-time bearer token which can be used to invoke selected integrator.io API routes.
+ *  
+ *  _integrationId - _id of the integration.
+ *
+ *  version - this field contains the version of the integration installed.
+ *
+ */
+
+module.settings.getMappingMetadata = function (options, callback) {
+  /*
+   *  settings function code
+   */
+
+  /*
+   *  err - return Error object but not handled in IO UI.
+   *       
+   *  response - Response data is an array of JSON objects where each object should follow following structure:
+   *  {"import1-externalId":[{"requiredGenerateFields":["generateFieldId1", "generateFieldId2"...],"nonEditableGenerateFields":[""]},{"generateList":"generateListId1","requiredGenerateFields":[""],"nonEditableGenerateFields":[""]}], "import2-externalId" : [...]} 
+   * 
+   * externalId - Import adaptor external Id. 
+   * 
+   * requiredGenerateFields - A collection of generate field ids that should be marked as required on UI.
+   * 
+   * nonEditableGenerateFields - A collection of generate field ids that should be marked as non-editable on UI. 
+   * 
+   * generateList - It contains generate list id of sublist and used to make the mapping mandatory of the sublist.
+   * 
+   * If mappings are required on both import and export side then you should add mapping generate field id in both requiredGenerateFields and nonEditableGenerateFields. 
```

Comment:
instead of "export and import side", lets write extracts and generates at here and below from an import perspective, as this metadata is for import. Write something like following 


Above properties can be used in following way : 
 -> To ensure an extract-generate mapping pair cannot be altered, add the generate field id both requiredGenerateFields and nonEditableGenerateFields. 
-> To ensure that a generate mapping is always present and mapped with any extract/hardcoded value, add the that generate field id to requiredGenerateFields.